### DISCUSSION TITLE
Geomap: Position is only displayed with MERGE

### DISCUSSION THREAD

**Discussion by UlrichThiess** [grafana/grafana#41507] [Dashboards, Panels & Visualizations]
Grafana-8.1.0-29738pre
on Windows Server 2019

Although Latitude and Longitude are available for selection in the fields under Datalayer, the position is only displayed if under Tranform Merge is selected.

Latitude and longitude are each saved separately and must therefore be queried with a separate request.

![grafik](https://user-images.githubusercontent.com/79257887/127441530-83d39304-e2f1-4b01-854d-f54b77b27c41.png)

![grafik](https://user-images.githubusercontent.com/79257887/127441569-6f4479ca-c1fd-4902-ab9f-e44a6d3408f9.png)

**Comment by ryantxu** [comment]
Hymm -- is there anything wrong with using the explicit merge if your query results return multiple frames?

Alternativly, can't influx return both lat and lon in the same query?  or do you need it to be from two queries?


For sure we should improve hits and feedback that the 'latitude' and 'longitude' field where not found in the same frame!

**Comment by UlrichThiess** [comment]
I'm using a Cerbo GX from Victron Energy here. This writes the data in two tables on the Influx SQL server. So I have two queries. One for longitude and one for latitude.

![grafik](https://user-images.githubusercontent.com/79257887/130048349-17446636-8600-4a4f-8dde-a842b8d536df.png)

Now you can select the appropriate values for `Latitude` and `Longitude`. I was confused when it didn't work. Only when I selected `Merge` as` Transform` the coordinates were displayed.

Maybe I can improve my Influx query?

**Comment by thinkingerrol** [comment]
The worst thing about this issue is of course the lack of an error message. I was super confused, why my data points would not be displayed until I found this issue. But I had to add a "Concatenate fields" transformation in the Transform tab, "Merge" would not work for me. Nice thing is there is the "Table view" toggle at the top, which allows to see what the current transformation (or just the query results) look like in raw form.

**Comment by ryantxu** [comment]
Interesting -- lets leave this as a discussion thread while we figure out what makes sense.  I am reluctant to automatically pull fields from multiple frames to create location values.  That seems much more like something you want to do explicitly

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
